File: tests/src/main/java/com/github/klee0kai/test/mowgli/earth/Cave.java
package com.github.klee0kai.test.mowgli.earth;

import java.util.UUID;


public class Cave implements IMountain {

    public UUID uuid = UUID.randomUUID();

    public Cave() {
        //nothing
    }

    public Cave(CaveType type, int deep) {
        //nothing
    }

    public  enum CaveType {
        Solutional, Glacier, Fracture
    }

}

File: tests/src/main/java/com/github/klee0kai/test/mowgli/earth/Cave.kt
package com.github.klee0kai.test.mowgli.earth

import java.util.*

class Cave : IMountain {
    var uuid: UUID = UUID.randomUUID()

    constructor()

    constructor(type: CaveType?, deep: Int?)

    enum class CaveType {
        Solutional, Glacier, Fracture
    }
}
